### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by Ian-Stewart-Binks**
Just some cleanup:
* Removed unused JS variables
* Removed unused JS functions/code
* Removed redundant HS code
* Renamed/factored out some logic in sidebar code
* Renamed _public/js/search/timetable.js_ to conform to new naming convention

### CODE REVIEW

**Comment by christinem** [comment]
I don't actually think we talked about naming jsx files <file>.js.jsx, unless I missed something. @david-yz-liu do you want to use this convention?

**Comment by Ian-Stewart-Binks** [comment]
[Scroll down to David's comment on jsx trend](https://github.com/Courseography/courseography/pull/719)

**Comment by christinem** [comment]
@Ian-Stewart-Binks ah, got it, thanks!

**Comment by Ian-Stewart-Binks** [comment]
Still good to get feedback on it from David though. Thanks for taking a look, @christinem!

File: hs/MasterTemplate.hs [hunk 1]
```diff
@@ -20,9 +20,9 @@ masterTemplate title headers body scripts =
             H.title (H.toHtml title)
             H.link ! A.rel "icon" ! A.type_ "image/png"
                    ! A.href "static/res/ico/favicon.png"
-            sequence_ $ headers ++ (map toStylesheet [
```
Comment (david-yz-liu) [inline hs/MasterTemplate.hs]: Could this part be simplified to something like

```
headers
mapM_ toStylesheet [ ...]
```

**Comment by david-yz-liu** [comment]
@Ian-Stewart-Binks Looks great. I made wrote one small question. Thanks for doing this!

File: hs/MasterTemplate.hs (hunk 1 above)
Comment (Ian-Stewart-Binks) [inline hs/MasterTemplate.hs]: That would make an awful lot more sense. Thanks for the suggestion.

**Comment by david-yz-liu** [comment]
Thanks, @Ian-Stewart-Binks!

**Pull request merged**